@here/harp-mapview/lib/text/TextElementGroupState.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { TextElement } from "./TextElement";
+import { TextElementGroup } from "./TextElementGroup";
+import { TextElementState } from "./TextElementState";
+
+/**
+ * Type of functions used to do early rejection of elements during group state creation or update.
+ * @param textElement The text element to check.
+ * @param lastFrameNumber Last frame the element was processed for rendering if available, otherwise
+ * `undefined`.
+ * @returns `undefined` if element was rejected, otherwise its current view distance.
+ */
+export type TextElementFilter = (
+    textElement: TextElement,
+    lastFrameNumber?: number
+) => number | undefined;
+
+/**
+ * `TextElementGroupState` keeps the state of a text element group while it's being rendered.
```

Comment:
It might be worth being a bit more explicit, i.e. it keeps the state of each text element in the group as well.

Reply:
done.